tools/PTE/pte-snapshot.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright 2016 IBM All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ *   usage:
+ *    node pte-snapshot.js <Nid> <uiFile> <org> <PTEid>
+ *       - Nid: Network id
+ *       - uiFile: user input file
+ *       - org: organization
+ *       - PTEid: PTE id
+ */
+'use strict';
+
+let hfc = require('fabric-client');
+let fs = require('fs');
+let testUtil = require('./pte-util.js');
+
+// inputs
+let Nid = parseInt(process.argv[2]);
+let uiFile = process.argv[3];
+let org = process.argv[4];
+let PTEid = parseInt(process.argv[5]);
+PTEid = PTEid ? PTEid : 0
+let loggerMsg = 'PTE ' + PTEid + ' snapshot';
+let logger = new testUtil.PTELogger({ "prefix": loggerMsg, "level": "info" });
+logger.info('input parameters: Nid=%d, uiFile=%s, oeg=%s, PTEid=%d', Nid, uiFile, org, PTEid);
+
+let txCfgPtr;
+let txCfgTmp;
+let uiContent;
+if (fs.existsSync(uiFile)) {
+    uiContent = testUtil.readConfigFileSubmitter(uiFile);
+
+    if (typeof (uiContent.txCfgPtr) === 'undefined') {
+        txCfgTmp = uiFile;
+    } else {
+        txCfgTmp = uiContent.txCfgPtr;
+    }
+    txCfgPtr = testUtil.readConfigFileSubmitter(txCfgTmp);
+}
+else {
+    uiContent = JSON.parse(uiFile)
+    txCfgPtr = uiContent;
+}
+
+let TLS = testUtil.setTLS(txCfgPtr);
+
+// snapshot input
+if (typeof (txCfgPtr.snapshot) === 'undefined') {
+    logger.error('snapshot input data is required');
+    process.exit(1);
+}
+let snapshot = txCfgPtr.snapshot;
+logger.info('snapshot: %j', snapshot);
+
+let queryFreq = 10000;        // default query info frequency = 10 sec
+if (typeof (snapshot.queryFreq) !== 'undefined') {
+    queryFreq = parseInt(snapshot.queryFreq);
+}
+logger.info('snapshot query info frequency: %d', queryFreq);
+
+// find the connection profile that contains the org
+let cpList = [];
+let cpPath = uiContent.ConnProfilePath;
+cpList = testUtil.getConnProfileListSubmitter(cpPath);
+if (cpList.length === 0) {
+    logger.error('error: invalid connection profile path or no connection profiles found in the connection profile path: %s', cpPath);
+    process.exit(1);
+}
+
+let cpf = testUtil.findOrgConnProfileSubmitter(cpList, org);
+if (null === cpf) {
```

Comment:
`if (!cpf)`

Null evaluates to `false` in Node.js (one of the few things I actually like about Node.js)